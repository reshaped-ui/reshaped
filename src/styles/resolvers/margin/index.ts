import { responsiveVariables } from "utilities/props";
import * as T from "styles/types";
import "./margin.css";

const margin: T.StyleResolver<T.Margin> = (value) => {
	if (!value) return {};
	return { variables: responsiveVariables("--rs-m", value) };
};

export const marginTop: T.StyleResolver<T.Margin> = (value) => {
	if (value === undefined) return {};
	return { variables: responsiveVariables("--rs-m-top", value) };
};

export const marginBottom: T.StyleResolver<T.Margin> = (value) => {
	if (value === undefined) return {};
	return { variables: responsiveVariables("--rs-m-bottom", value) };
};

export const marginStart: T.StyleResolver<T.Margin> = (value) => {
	if (value === undefined) return {};
	return { variables: responsiveVariables("--rs-m-start", value) };
};

export const marginEnd: T.StyleResolver<T.Margin> = (value) => {
	if (value === undefined) return {};
	return { variables: responsiveVariables("--rs-m-end", value) };
};

export const marginInline: T.StyleResolver<T.Margin> = (value) => {
	if (value === undefined) return {};
	return { variables: responsiveVariables("--rs-m-inline", value) };
};

export const marginBlock: T.StyleResolver<T.Margin> = (value) => {
	if (value === undefined) return {};
	return { variables: responsiveVariables("--rs-m-block", value) };
};

export default margin;
