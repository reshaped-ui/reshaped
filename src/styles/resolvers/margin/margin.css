@responsive [style*="--rs-m-s"] {
	@variable --rs-m 0;

	margin: calc(var(--rs-m) * var(--rs-unit-x1));
}

@responsive [style*="--rs-m-top-"] {
	@variable --rs-m-top 0;

	margin-top: calc(var(--rs-m-top) * var(--rs-unit-x1));
}

@responsive [style*="--rs-m-bottom-"] {
	@variable --rs-m-bottom 0;

	margin-bottom: calc(var(--rs-m-bottom) * var(--rs-unit-x1));
}

@responsive [style*="--rs-m-start-"] {
	@variable --rs-m-start 0;

	margin-inline-start: calc(var(--rs-m-start) * var(--rs-unit-x1));
}

@responsive [style*="--rs-m-end-"] {
	@variable --rs-m-end 0;

	margin-inline-end: calc(var(--rs-m-end) * var(--rs-unit-x1));
}

@responsive [style*="--rs-m-block-"] {
	@variable --rs-m-block 0;

	margin-block: calc(var(--rs-m-block) * var(--rs-unit-x1));
}

@responsive [style*="--rs-m-inline-"] {
	@variable --rs-m-inline 0;

	margin-inline: calc(var(--rs-m-inline) * var(--rs-unit-x1));
}
