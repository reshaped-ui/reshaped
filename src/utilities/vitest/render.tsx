import { render as vitestRender } from "vitest-browser-react";
import Reshaped from "components/Reshaped";
import "themes/reshaped/theme.css";

const Wrapper = (props: { children: React.ReactNode }) => {
	return <Reshaped>{props.children}</Reshaped>;
};

const render = (node: React.ReactNode) => {
	const screen = vitestRender(<div id="root">{node}</div>, {
		wrapper: Wrapper,
	});

	return { ...screen, container: document.getElementById("root")! };
};

export default render;
