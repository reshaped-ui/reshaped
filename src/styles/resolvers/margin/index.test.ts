import { expect, test, describe } from "vitest";
import margin, {
	marginBlock,
	marginBottom,
	marginEnd,
	marginInline,
	marginStart,
	marginTop,
} from "./index";

describe("Styles/Margin", () => {
	test("handles positive value", () => {
		expect(margin(4)).toMatchSnapshot();
	});

	test("handles 0 value", () => {
		expect(margin(0)).toMatchSnapshot();
	});

	test("handles undefined value", () => {
		expect(margin()).toMatchSnapshot();
	});

	test("handles responsive value", async () => {
		expect(margin({ s: 4, m: 0, l: 2 })).toMatchSnapshot();
	});
});

describe("Styles/Margin/Top", () => {
	test("handles positive value", () => {
		expect(marginTop(4)).toMatchSnapshot();
	});

	test("handles 0 value", () => {
		expect(marginTop(0)).toMatchSnapshot();
	});

	test("handles undefined value", () => {
		expect(marginTop()).toMatchSnapshot();
	});

	test("handles responsive value", async () => {
		expect(marginTop({ s: 4, m: 0, l: 2 })).toMatchSnapshot();
	});
});

describe("Styles/Margin/Bottom", () => {
	test("handles positive value", () => {
		expect(marginBottom(4)).toMatchSnapshot();
	});

	test("handles 0 value", () => {
		expect(marginBottom(0)).toMatchSnapshot();
	});

	test("handles undefined value", () => {
		expect(marginBottom()).toMatchSnapshot();
	});

	test("handles responsive value", async () => {
		expect(marginBottom({ s: 4, m: 0, l: 2 })).toMatchSnapshot();
	});
});

describe("Styles/Margin/Start", () => {
	test("handles positive value", () => {
		expect(marginStart(4)).toMatchSnapshot();
	});

	test("handles 0 value", () => {
		expect(marginStart(0)).toMatchSnapshot();
	});

	test("handles undefined value", () => {
		expect(marginStart()).toMatchSnapshot();
	});

	test("handles responsive value", async () => {
		expect(marginStart({ s: 4, m: 0, l: 2 })).toMatchSnapshot();
	});
});

describe("Styles/Margin/End", () => {
	test("handles positive value", () => {
		expect(marginEnd(4)).toMatchSnapshot();
	});

	test("handles 0 value", () => {
		expect(marginEnd(0)).toMatchSnapshot();
	});

	test("handles undefined value", () => {
		expect(marginEnd()).toMatchSnapshot();
	});

	test("handles responsive value", async () => {
		expect(marginEnd({ s: 4, m: 0, l: 2 })).toMatchSnapshot();
	});
});

describe("Styles/Margin/Inline", () => {
	test("handles positive value", () => {
		expect(marginInline(4)).toMatchSnapshot();
	});

	test("handles 0 value", () => {
		expect(marginInline(0)).toMatchSnapshot();
	});

	test("handles undefined value", () => {
		expect(marginInline()).toMatchSnapshot();
	});

	test("handles responsive value", async () => {
		expect(marginInline({ s: 4, m: 0, l: 2 })).toMatchSnapshot();
	});
});

describe("Styles/Margin/Block", () => {
	test("handles positive value", () => {
		expect(marginBlock(4)).toMatchSnapshot();
	});

	test("handles 0 value", () => {
		expect(marginBlock(0)).toMatchSnapshot();
	});

	test("handles undefined value", () => {
		expect(marginBlock()).toMatchSnapshot();
	});

	test("handles responsive value", async () => {
		expect(marginBlock({ s: 4, m: 0, l: 2 })).toMatchSnapshot();
	});
});
