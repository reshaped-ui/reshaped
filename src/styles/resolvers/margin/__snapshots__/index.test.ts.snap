// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Styles/Margin > handles 0 value 1`] = `{}`;

exports[`Styles/Margin > handles positive value 1`] = `
{
  "variables": {
    "--rs-m-s": 4,
  },
}
`;

exports[`Styles/Margin > handles responsive value 1`] = `
{
  "variables": {
    "--rs-m-l": 2,
    "--rs-m-m": 0,
    "--rs-m-s": 4,
  },
}
`;

exports[`Styles/Margin > handles undefined value 1`] = `{}`;

exports[`Styles/Margin/Block > handles 0 value 1`] = `
{
  "variables": {
    "--rs-m-block-s": 0,
  },
}
`;

exports[`Styles/Margin/Block > handles positive value 1`] = `
{
  "variables": {
    "--rs-m-block-s": 4,
  },
}
`;

exports[`Styles/Margin/Block > handles responsive value 1`] = `
{
  "variables": {
    "--rs-m-block-l": 2,
    "--rs-m-block-m": 0,
    "--rs-m-block-s": 4,
  },
}
`;

exports[`Styles/Margin/Block > handles undefined value 1`] = `{}`;

exports[`Styles/Margin/Bottom > handles 0 value 1`] = `
{
  "variables": {
    "--rs-m-bottom-s": 0,
  },
}
`;

exports[`Styles/Margin/Bottom > handles positive value 1`] = `
{
  "variables": {
    "--rs-m-bottom-s": 4,
  },
}
`;

exports[`Styles/Margin/Bottom > handles responsive value 1`] = `
{
  "variables": {
    "--rs-m-bottom-l": 2,
    "--rs-m-bottom-m": 0,
    "--rs-m-bottom-s": 4,
  },
}
`;

exports[`Styles/Margin/Bottom > handles undefined value 1`] = `{}`;

exports[`Styles/Margin/End > handles 0 value 1`] = `
{
  "variables": {
    "--rs-m-end-s": 0,
  },
}
`;

exports[`Styles/Margin/End > handles positive value 1`] = `
{
  "variables": {
    "--rs-m-end-s": 4,
  },
}
`;

exports[`Styles/Margin/End > handles responsive value 1`] = `
{
  "variables": {
    "--rs-m-end-l": 2,
    "--rs-m-end-m": 0,
    "--rs-m-end-s": 4,
  },
}
`;

exports[`Styles/Margin/End > handles undefined value 1`] = `{}`;

exports[`Styles/Margin/Inline > handles 0 value 1`] = `
{
  "variables": {
    "--rs-m-inline-s": 0,
  },
}
`;

exports[`Styles/Margin/Inline > handles positive value 1`] = `
{
  "variables": {
    "--rs-m-inline-s": 4,
  },
}
`;

exports[`Styles/Margin/Inline > handles responsive value 1`] = `
{
  "variables": {
    "--rs-m-inline-l": 2,
    "--rs-m-inline-m": 0,
    "--rs-m-inline-s": 4,
  },
}
`;

exports[`Styles/Margin/Inline > handles undefined value 1`] = `{}`;

exports[`Styles/Margin/Start > handles 0 value 1`] = `
{
  "variables": {
    "--rs-m-start-s": 0,
  },
}
`;

exports[`Styles/Margin/Start > handles positive value 1`] = `
{
  "variables": {
    "--rs-m-start-s": 4,
  },
}
`;

exports[`Styles/Margin/Start > handles responsive value 1`] = `
{
  "variables": {
    "--rs-m-start-l": 2,
    "--rs-m-start-m": 0,
    "--rs-m-start-s": 4,
  },
}
`;

exports[`Styles/Margin/Start > handles undefined value 1`] = `{}`;

exports[`Styles/Margin/Top > handles 0 value 1`] = `
{
  "variables": {
    "--rs-m-top-s": 0,
  },
}
`;

exports[`Styles/Margin/Top > handles positive value 1`] = `
{
  "variables": {
    "--rs-m-top-s": 4,
  },
}
`;

exports[`Styles/Margin/Top > handles responsive value 1`] = `
{
  "variables": {
    "--rs-m-top-l": 2,
    "--rs-m-top-m": 0,
    "--rs-m-top-s": 4,
  },
}
`;

exports[`Styles/Margin/Top > handles undefined value 1`] = `{}`;
